From c86b9af52afb9fbcdbe238626b8f2004813ba0d7 Mon Sep 17 00:00:00 2001
From: Mikael Arguedas <mikael.arguedas@gmail.com>
Date: Fri, 15 May 2015 22:00:21 +0200
Subject: [PATCH] convert CMake to qibuild

---
 CMakeLists.txt           | 17 ++++++++++++-----
 libqicore/CMakeLists.txt | 14 ++++++--------
 2 files changed, 18 insertions(+), 13 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4b6af3b..6aa917f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,14 +1,21 @@
 ## Copyright (c) 2012 Aldebaran Robotics. All rights reserved.
 
 cmake_minimum_required(VERSION 2.8)
-project(LibQiCore)
-find_package(qibuild)
-qi_sanitize_compile_flags(HIDDEN_SYMBOLS)
+project(naoqi_libqicore)
+find_package(catkin REQUIRED naoqi_libqi)
+find_package(Boost)
+
+catkin_package(LIBRARIES qicore
+  INCLUDE ./
+  DEPENDS Boost
+)
 
 # Tests
 enable_testing()
 
 add_subdirectory("libqicore")
-add_subdirectory("services")
-add_subdirectory("bin")
+#add_subdirectory("services")
+#add_subdirectory("bin")
 
+# Install catkin package.xml
+install(FILES package.xml DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION})
diff --git a/libqicore/CMakeLists.txt b/libqicore/CMakeLists.txt
index b01d020..fcf0f67 100644
--- a/libqicore/CMakeLists.txt
+++ b/libqicore/CMakeLists.txt
@@ -1,6 +1,5 @@
 ## Copyright (c) 2012, 2014 Aldebaran Robotics. All rights reserved.
 
-find_package(qimodule)
 include_directories(".")
 
 set(PUBLIC_HEADERS
@@ -11,20 +10,19 @@ set(PUBLIC_HEADERS
     qicore/logmessage.hpp
     qicore/logprovider.hpp
     )
-qi_install_header(${PUBLIC_HEADERS})
+install(FILES ${PUBLIC_HEADERS} DESTINATION ${CATKIN_GLOBAL_INCLUDE_DESTINATION}/qicore)
 
-qi_create_module(qicore SRC
+add_library(qicore
   ${PUBLIC_HEADERS}
   #src/behavior.cpp src/behaviormodel.cpp src/format_behavior_model.cpp
   src/loglistener_proxy.cpp
   src/logmanager_proxy.cpp
   src/logprovider_proxy.cpp
-  src/registration.cpp
   src/logproviderimpl.cpp
   src/logproviderimpl.hpp
-  DEPENDS BOOST
 )
-qi_use_lib(qicore QI)
-qi_stage_lib(qicore)
 
-add_subdirectory("tests")
+target_link_libraries(qicore ${Boost_LIBRARIES} ${catkin_LIBRARIES})
+install(TARGETS qicore DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION})
+
+#add_subdirectory("tests")
-- 
1.9.1

